add strict meta descriptor

cyx opened the pull request:
This change adds the ability to flag an env var as `strict`.
Strict env vars will throw an error if parsing fails.

In all honesty, this is a good default to have, but given that
it's a breaking change, opting to make it optional is probably
the more prudent change.

joeshaw commented:
This code looks good.  Thank you for the patch.

Would you mind adding documentation about this setting to the `Decode` function?

I agree with you that throwing errors ought to be the default, and mention this in #9.  The only reason why it doesn't is for backward compatibility, and if I were starting again I'd probably change this.  I think this is a good balance and can resolve #9 when it lands.

Another potential addition would be a `DecodeStrict` function in which this was the default.

cyx commented:
@joeshaw I've just updated everything, and went with `StrictDecode` as the function name. I can rename if you think the other way is better.

Also added README.

joeshaw commented:
Perfect! Thanks a lot!
